### PR TITLE
Update FocusEvent pages

### LINKED ISSUE #14084: Content bug: Description of focus events matches current spec but not reality
## What page(s) did you find the problem on?

* [focus event](https://developer.mozilla.org/en-US/docs/Web/API/Element/focus_event)
* [blur event](https://developer.mozilla.org/en-US/docs/Web/API/Element/blur_event)
* [focusin event](https://developer.mozilla.org/en-US/docs/Web/API/Element/focusin_event)
* [focusout event](https://developer.mozilla.org/en-US/docs/Web/API/Element/focusout_event)

## What is the problem?

The pages match [the text of the current spec](https://www.w3.org/TR/uievents/#events-focus-types), but browsers uniformly implement a different story, and it looks like [the spec is due for an update](https://github.com/w3c/uievents/pull/290) to match the implementations.

## What did you expect to see?

Should we be proactive and update the descriptions to match the soon-to-be-updated spec / current browser implementations?

### PR DESCRIPTION
**Pull request by danburzo**
#### Summary

I've made the following updates: 

* Note that focusin/focusout follow their non-bubbling counterparts;
* Note that focusin/focusout don't (yet) have an event handling property;
* Add the description of the FocusEvent.relatedTarget property on all events;
* Link to HTMLElement.focus() / .blur() methods from the similarly-named event pages.
* Adds a section to FocusEvent that describes the difference in the order of events between the spec and current browser implementations, and points to it from the focusin/focusout pages.

Fixes #14084. 

#### Motivation

I propose these changes to make the FocusEvent pages more useful to readers by matching what they will experience in browsers.

#### Supporting details

* [UIEvents spec: focus event order](https://w3c.github.io/uievents/#events-focusevent-event-order)
* https://github.com/w3c/uievents/issues/88
* https://github.com/whatwg/html/issues/3514
* https://github.com/whatwg/html/issues/4542

#### Related issues

Fixes #14084. 

#### Metadata

- [ ] Adds a new document
- [x] Rewrites (or significantly expands) a document
- [x] Fixes a typo, bug, or other error

### CODE REVIEW

**Review by jpmedley — CHANGES_REQUESTED** [review]

File: files/en-us/web/api/element/blur_event/index.md [hunk 1]
```diff
@@ -46,6 +46,10 @@ The opposite of `blur` is {{domxref("Element/focus_event", "focus")}}.
       <th scope="row">Composed</th>
       <td>Yes</td>
     </tr>
+    <tr>
+      <th scope="row">{{DOMxRef("FocusEvent.relatedTarget")}}</th>
+      <td>The element receiving focus, if any.</td>
```
[the remark this answers is not in the document]
Comment (danburzo) [reply files/en-us/web/api/element/blur_event/index.md:51]: Yes, when the focus moves outside the web page — for example, if you were to open the DevTools and focus the Console prompt — the `focusout` event would have `relatedTarget: null`. Ditto when you click on an element on the page that's not focusable.  The [page for `relatedTarget`](https://developer.mozilla.org/en-US/docs/Web/API/FocusEvent/relatedTarget) documents these scenarios.

File: files/en-us/web/api/element/focus_event/index.md [hunk 2]
```diff
@@ -45,6 +45,10 @@ The opposite of `focus` is {{domxref("Element/blur_event", "blur")}}.
       <th scope="row">Composed</th>
       <td>Yes</td>
     </tr>
+    <tr>
+      <th scope="row">{{DOMxRef("FocusEvent.relatedTarget")}}</th>
+      <td>The element losing focus, if any.</td>
```
[the remark this answers is not in the document]
Comment (danburzo) [reply files/en-us/web/api/element/focus_event/index.md:50]: Similarly to the scenarios described above, when the page first gets a focused element, or you return the focus from outside the page, `relatedTarget` will be null.

File: files/en-us/web/api/element/focusin_event/index.md [hunk 3]
```diff
@@ -45,6 +43,10 @@ The opposite of `focusin` is {{domxref("Element/focusout_event", "focusout")}}.
       <th scope="row">Composed</th>
       <td>Yes</td>
     </tr>
+    <tr>
+      <th scope="row">{{DOMxRef("FocusEvent.relatedTarget")}}</th>
+      <td>The element losing focus, if any.</td>
```
[the remark this answers is not in the document]
Comment (danburzo) [reply files/en-us/web/api/element/focusin_event/index.md:48]: Same answer as for the `focus` event.

File: files/en-us/web/api/element/focusout_event/index.md [hunk 4]
```diff
@@ -46,6 +44,10 @@ The opposite of `focusout` is {{domxref("Element/focusin_event", "focusin")}}.
       <th scope="row">Composed</th>
       <td>Yes</td>
     </tr>
+    <tr>
+      <th scope="row">{{DOMxRef("FocusEvent.relatedTarget")}}</th>
+      <td>The element receiving focus, if any.</td>
```
[the remark this answers is not in the document]
Comment (danburzo) [reply files/en-us/web/api/element/focusout_event/index.md:49]: Same answer as for the `blur` event.

File: files/en-us/web/api/element/blur_event/index.md (hunk 1 above)
Comment (jpmedley) [inline files/en-us/web/api/element/blur_event/index.md:51]:
```suggestion
      <td>The element receiving focus, if any. (Follow the link for details.)</td>
```

**Review by jpmedley — CHANGES_REQUESTED** [review]

File: files/en-us/web/api/element/blur_event/index.md (hunk 1 above)
Comment (danburzo) [reply files/en-us/web/api/element/blur_event/index.md:51]: How about:

... | ...
--- | ---
Secondary target | [`relatedTarget`](https://developer.mozilla.org/en-US/docs/Web/API/FocusEvent/relatedTarget) points to the element receiving focus, if any.

Or, more explicitly:

... | ...
--- | ---
Secondary target | The element receiving focus, if any. (See [`Focus.relatedTarget`](https://developer.mozilla.org/en-US/docs/Web/API/FocusEvent/relatedTarget) for details.)

(I'm trying to bring the link closer to the instructions to follow it.)

Comment (danburzo) [reply files/en-us/web/api/element/blur_event/index.md]: I went for the latter.

**Comment by Rumyra** [comment]
@danburzo from what I can see the changes suggested have been made - can you confirm?

If you resolve conflicts we should be able to get this pr through 👍

**Comment by danburzo** [comment]
@Rumyra I've adapted my proposed changes to the new structure for FocusEvent pages. 

Substantial new things:

- The description for `FocusEvent.relatedTarget` is now specific to each event, but does not repeat the cases where `relatedTarget` might be null, which can be consulted on the property's page.
- I've removed mentions of the `onfocusin` / `onfocusout` event handling properties, which are not currently specced (see [whatwg/html#4542](https://github.com/whatwg/html/issues/4542))

**Comment by estelle** [comment]
@jpmedley Is this ok to merge?

**Review by jpmedley — APPROVED** [review]

**Pull request merged**